src/librbd/plugin/Types.h

```diff
@@ -14,8 +15,43 @@ namespace plugin {
 
 template <typename> struct Api;
 
+template <typename ImageCtxT>
+struct HookPointInterface {
+  virtual void open(ImageCtxT* image_ctx, Context* on_finish) = 0;
```

Comment:
Nit: these first two (`open` and `close`) are really `acquire_exclusive_lock` and `release_exclusive_lock`

Reply:
Renamed.